modules/consensus/block_validation.go

```diff
@@ -63,7 +64,7 @@ func checkTarget(b types.Block, target types.Target) bool {
 // ValidateBlock validates a block against a minimum timestamp, a block target,
 // and a block height. Returns nil if the block is valid and an appropriate
 // error otherwise.
-func (bv stdBlockValidator) ValidateBlock(b types.Block, minTimestamp types.Timestamp, target types.Target, height types.BlockHeight) error {
+func (bv stdBlockValidator) ValidateBlock(b types.Block, minTimestamp types.Timestamp, target types.Target, height types.BlockHeight, log *persist.Logger) error {
```

Comment:
I wouldn't worry about that too much, especially if you wrap the block in `if build.DEBUG` so that you don't incur the encoding cost unless you're running debug.

On that note, I think it might be worthwhile to add a `EncodedLen` function to the `encoding` package. It could just count bytes instead of actually writing anything, so you wouldn't need to allocate memory. I'm not sure how common `len(encoding.Marshal(x))` is, though.